### DISCUSSION TITLE
Strange name recognition issue

### DISCUSSION THREAD

**Discussion by the1gofer** [explosion/spaCy#8326] [Help: Other Questions] [answered]
I have the following sentence:

```In addition to his parents, he was preceded in death by 2-brothers:  Dennis R. Crews and Jerry Mitchell Crews.```

When I get the names using the web_trf model, it finds two names, Dennis R. Crews and Jerry Mitchell Crews, as expected.

However when  do the same thing with the following sentences,  Dennis R. Crews becomes  R. Crews.   

```
Alton Dale Crews, 66, of Starke, FL passed away Monday, March 1, 2021 in Jacksonville, FL.  He was born April 25, 1954 in Folkston, GA to the late Chester Thomas Crews and Christine Geiger Crews.  He retired after 25 years as a car hauler.  He was a member of the Homeland Pentecostal Church where he had served as pastor for the last 17 years.

In addition to his parents, he was preceded in death by 2-brothers:  Dennis R. Crews and Jerry Mitchell Crews.

He is survived by his wife of 47 years:  Kathy Lightsey Crews; 2-children:  Alton Crews, Jr and Wendi Crews; 2-brothers:  Floyd T. (Linda) Crews and D. Steven Crews; a sister:  Sheila Crews Brandies; 4-grandchildren:  Tyler Ray Conner, Gavin Del Crews, Magnolia Dale Carroll and Hunter Ray Crews  and 2-great grandchildren:  Brayden Isaac Carroll and Sawyer Isaiah Carroll.
```

Any idea why?

**Comment by polm** [comment, marked as answer]
Honestly no idea, sometimes the models just work in mysterious ways. I'll link in https://github.com/explosion/spaCy/issues/3052 since it's relevant here.

One thing is that because this seems to be an effect of a pretty large context, you might have more luck with non-Transformers models. 

(I assume "sounding" was a typo for "following", but if not let me know.)

**Reply by the1gofer** [reply]
I've found it's the most accurate for the names in general, is their a model that's more accurate for names?

**Reply by polm** [reply]
We do not currently distribute any special-purpose models such as one for names. 

In general the Transformers models is going to be the best. However, as mentioned in the linked thread, a model that's 99% accurate is going to be wrong 1% of the time, and you seem to have gotten unlucky with the above text. 

If coverage is the most important thing for your application, you might consider running multiple models and combining their results.

**Comment by the1gofer** [comment]
It seems to be fine If I extract the names sentence by sentence so that’s
my approach at the moment.

On Fri, Jun 11, 2021 at 9:41 PM polm ***@***.***> wrote:

> We do not currently distribute any special-purpose models such as one for
> names.
>
> In general the Transformers models is going to be the best. However, as
> mentioned in the linked thread, a model that's 99% accurate is going to be
> wrong 1% of the time, and you seem to have gotten unlucky with the above
> text.
>
> If coverage is the most important thing for your application, you might
> consider running multiple models and combining their results.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/explosion/spaCy/discussions/8326#discussioncomment-860727>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAQKUNDTJPHIZTKDEQILV3DTSLQQDANCNFSM46M4JTZQ>
> .
>
-- 
Jason Crews
